website/pages/docs/nia/installation/requirements.mdx

```diff
@@ -40,8 +42,104 @@ The modules listed below are availabe to use and are comptabible with consul-ter
+| `node_id` | ID of the node on which the service is registered. |
+| `node_address` | The IP address of the Consul node on which the service is registered. |
+| `node_datacenter` | Data center of the Consul node on which the service is registered. |
+| `node_tagged_addresses` | List of explicit LAN and WAN IP addresses for the agent |
+| `node_meta` | List of user-defined metadata key/value pairs for the node |
+
+Terraform variables when passed as module arguments can be [lossy for object types](https://www.terraform.io/docs/configuration/types.html#conversion-of-complex-types). This allows Consul-Terraform-Sync to declare the full variable with every object attribute in the generated root module, and pass the variable to a child module that contains a subset of these attributes for its variable declaration. Modules compatible with Consul-Terraform-Sync may simplify the services variable within the module by omitting unused attributes. For example, the following services variable has 4 attributes with the rest omittied.
+
+```hcl
+variable "services" {
+  description = "Consul services monitored by Consul-Terraform-Sync"
+  type = map(
+    object({
+      id           = string
+      name         = string
+      node_address = string
+      port         = number
+      status       = string
+    })
+  )
+}
+```
+
+#### Module Input Variables
+
+In addition to the required `services` input variable, Consul-Terraform-Sync supports all input variables exposed by the module.
+
+Network infrastructure differs vastly across teams and organizations, and the automation needs of practitioners are unique based on their existing setup. Identify areas in the module where practitioners could tailor the automation to fit their infrastructure. Declare [input variables](https://www.terraform.io/docs/configuration/variables.html) and insert the use of variables throughout module resources to expose these options to practitioners. Include descriptions to capture what the variables are and how they are used, and specify [custom validation rules for variables](https://www.terraform.io/docs/configuration/variables.html#custom-validation-rules) to provide context to users the expected format and conditions for the variables. Set reasonable default values for variables that are optional, and omit default values for variables that are required module arguments.
+
+Terraform is an explicit configuration language and requires variables to be declared, typed, and passed explicitly through as module arguments. Consul-Terraform-Sync abstracts this by creating intermediate variables at the root level from values intended for the module. Value assignments are parsed to interpolate the corresponding variable declaration and are written to the appropriate Terraform files. A few assumptions are made for the intermediate variables: the variables users provide Consul-Terraform-Sync are declared and supported by the module, matching name and type.
+
+#### Module Guidelines
```

Comment:
Don't remember this section in my last review. I like it!

Reply:
ha surprise!! 🎉

thanks for the review

Reply:
Agree ✨